_posts/2023-02-17-jmsblog.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: "Blind SQL injection vulnerability in Jms Blog (jmsblog) PrestaShop module"
+categories: modules
+author:
+- Creabilis.com
+- TouchWeb.fr
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop"
+severity: "critical (9.8)"
+---
+
+The module Jms Blog (jmsblog) from Joommasters contains a Blind SQL injection vulnerability.
+This module is for the PrestaShop e-commerce platform and mainly provided with joo masters PrestaShop themes
+
+## Summary
+
+* **CVE ID**: To request
+* **Published at**: 2023-02-17
+* **Advisory source**: none
+* **Vendor**: PrestaShop
+* **Product**: jmsblog
+* **Impacted release**: at least 2.5.5 and 2.5.6
+* **Product author**: Joommasters
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+Several front controller in /controllers/front/ hold sensitives SQL calls that can be executed with a trivial http call and exploited to forge a blind SQL injection.
+
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
+
+## Possible malicious usage
+
+This vulnerability permits altering the shop’s database.
+
+## Patch
+
+```diff
+--- a/controllers/front/archive.php
++++ b/controllers/front/archive.php
+@@ -55,1 +55,1 @@ function getPosts
+-            ' AND DATE_FORMAT(hss.created,"%Y-%m") LIKE "%'.$_month.'%"
++            ' AND DATE_FORMAT(hss.created,"%Y-%m") LIKE "%'.pSQL($_month).'%"
+```
+
+```diff
+--- a/controllers/front/post.php
++++ b/controllers/front/post.php
+@@ -85,1 +85,1 @@ function getPosts
+-                WHERE pc.`email` = \''.$email.'\'
++                WHERE pc.`email` = \''.pSQL($email).'\'
+```
+
+```diff
+--- a/controllers/front/tag.php
++++ b/controllers/front/tag.php
+@@ -53,1 +53,1 @@ function getPosts
+-            ' AND hssl.`tags` LIKE "%'.$tag.'%"
++            ' AND hssl.`tags` LIKE "%'.pSQL($tag).'%"
+```
+
+## Timeline
+
+| Date | Action |
+|--|--|
+| 2022-09-01 | Issue discovered during a pentest |
+| 2023-02-17 | Contact the author |
+| 2023-02-17 | Publish this security advisory |
```

Comment:
| 2023-02-17 | Request a CVE ID |
| 2023-03-xx | Publish this security advisory |